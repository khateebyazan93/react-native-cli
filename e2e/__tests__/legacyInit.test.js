// @flow
import fs from 'fs';
import path from 'path';
import execa from 'execa';
import {run, getTempDirectory, cleanup, writeFiles} from '../helpers';

const DIR = getTempDirectory('command-legacy-init');

beforeEach(() => {
  cleanup(DIR);
  writeFiles(DIR, {});
});
afterEach(() => {
  cleanup(DIR);
});

test('legacy init through react-native-cli', () => {
  const templateFiles = [
    '.buckconfig',
    '.flowconfig',
    '.gitattributes',
    '.gitignore',
    '.watchmanconfig',
    'App.js',
    '__tests__',
    'android',
    'app.json',
    'babel.config.js',
    'index.js',
    'ios',
    'metro.config.js',
    'node_modules',
    'package.json',
    'yarn.lock',
  ];

  const {stdout} = execa.sync('npx', ['react-native-cli', 'init', 'TestApp'], {
    cwd: DIR,
  });

  expect(stdout).toContain('Run instructions');

  // make sure we don't leave garbage
  expect(fs.readdirSync(DIR)).toEqual(['TestApp']);
  expect(fs.readdirSync(path.join(DIR, 'TestApp'))).toEqual(templateFiles);

  const pkgJson = require(path.join(DIR, 'TestApp', 'package.json'));
  expect(pkgJson).toMatchSnapshot(
    'package.json contains necessary configuration',
  );
});
