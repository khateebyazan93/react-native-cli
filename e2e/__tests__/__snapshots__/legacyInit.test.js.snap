// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`legacy init through react-native-cli: package.json contains necessary configuration 1`] = `
Object {
  "dependencies": Object {
    "react": "16.8.3",
    "react-native": "0.59.2",
  },
  "devDependencies": Object {
    "@babel/core": "^7.4.0",
    "@babel/runtime": "^7.4.2",
    "@react-native-community/eslint-config": "^0.0.3",
    "babel-jest": "^24.5.0",
    "eslint": "^5.15.3",
    "jest": "^24.5.0",
    "metro-react-native-babel-preset": "^0.53.1",
    "react-test-renderer": "16.8.3",
  },
  "jest": Object {
    "preset": "react-native",
  },
  "name": "TestApp",
  "private": true,
  "scripts": Object {
    "lint": "eslint .",
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest",
  },
  "version": "0.0.1",
}
`;
